### PR TITLE
Add Storybook, a UI component explorer

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**
# Issue

Currently when building new pages and views we don't have good documentation practices around our Vue components. We also don't have a good way to share components with designers in order to build new pages following consistent design.

# Changes

Introducing [Storybook](https://storybook.js.org/). I highly recommend watching the video on the home page.
This tool allows us to develop pure UI components with a lot more confidence, and can guide more isolated component design.
It also provides examples of idiomatic usage of our components.

The majority of the work in this PR sets up a foundation for us to use `sass`, `pug` and `js` to build Vue components.

As `pug` needed to be updated to work within Vue components it's now advantageous to start migrating to using `pug`. If you reference a `jade` file make sure to provide the explicit file extension.
The build step has been updated to treat `.jade` and `.pug` files separately with `jade` and `pug` respectively. Thus there are now semantic differences between the two file types.

It is non trivial to update our `.jade` files to `.pug` due to breaking changes in pug. But `pug` can extend from `jade`. (Thank goodness)

This is also a step towards Webpack 4.

## Next steps

This PR does not include many of the opt-in addons that Storybook includes. I've only added the `storysource` addon so far to let us see the source code within Storybook. In the future we'll want to add a documentation addon that will allow us to add markdown with a story in order to provide additional documentation. We'll also probably want to add [Knobs](https://github.com/storybooks/storybook/tree/master/addons/knobs) which will allow a user to change props dynamically from the interface. Addon list [here](https://storybook.js.org/docs/addons/addon-gallery/).

We can then set up an automated script to deploy the storybook static site.

## Try it out

`npm i` and then run `npm run storybook`.
This will start a dev environment. Click on `PieComponent` story in the left menu or search for it.
Here you can see 3 examples of how to use this component.

Now create a new story or modify the percentage property in one of the pie components. Note the quick code reloading.

## Testing

As there are changes to the build process and view templates I've spent some extra time testing this branch.

I've tested this with our client tests but also deployed to staging and run browserstack smoke tests.

Therefore I have a high confidence that views are not broken.

## Punted work

~Fully upgrade from `jade` to `pug`, by upgrading `jade-loader` (which is a fork of `pug-loader`).~

~Changing `jade-loader` to `pug-loader` creates an issue with interpolation. [Reference](https://github.com/pugjs/pug-loader/issues/119). More work is required.~

~We prefer the file extensions to be `pug` as the `jade-loader` works on both `.pug` and `.jade`. The static file compilation with the newer `pug` requires `base-flat.pug` instead of `base-flat.jade`.~

I have integrated `pug-loader` with `jade-loader`. They handle their specific file.
This keeps us backward compatible and lets us move forward with modern `pug`.

The Vue components supported by storybook do not support `pug`. Currently the only additional support is for `sass` styles.

# Command Quick Reference

`npm run storybook` - Runs a Storybook server with hot reloading, giving you a UI development environment.

`npm run build-storybook` - Generates a static site of the Storybook within a directory called `storybook-static`. This can be used in the future to share our documentation with the broader team.

### CODE REVIEW

**Comment by AndrewJakubowicz** [comment]
To simplify the diffs I've tried to included minimal formatting in this PR.

**Comment by codeluggage** [comment]
Loving this! 

- The knobs plugin is a great idea. Not only does it make it much easier for us to see how the component works, but it lets the rest of the team interactively see how things work too. This could super charge our development cycle in terms of getting feedback and being on the same page.
- The static build could easily be combined with S3 hosting or serverless to quickly share in-development work with the rest of the team, specifically for that feedback.
- The only remaining `*.jade` files I find are in `bower_components/aether` 👍 
- Editing the stories updated super fast - really great dev experience. 

I did run into an issue with the `Educators` section of our site, though. All the links (typically `/teachers/something`) had an issue in their header. See attached screenshot: 

Screen Shot 2019-03-28 at 14 47 43

**Comment by shubhi1092** [comment]
Tested this today, looks great to test the vue components while moving towards isolated pure components!
I loved the extremely quick reloading after any changes, and the ability to test vue components in isolation. :)

**Comment by AndrewJakubowicz** [comment]
@codeluggage @shubhi1092 I've made changes based on your feedback.

Thank you so much for taking some time. I've also update PR description with new changes.

tl;dr:
 - pug-loader now added and handles `.pug` files. `jade-loader` remains for `.jade` files.
 - next steps (after merge) are to add Knobs and deploy automatically to static site.
